bench/compare-tree-performance.js

```diff
@@ -0,0 +1,65 @@
+const { Microcosm, update } = require('../build/microcosm')
+
+let repo = new Microcosm()
+
+let advance = (x, y) => ({ x, y })
+
+let SAMPLES = [25, 50, 100, 200]
+let WRITES = 50
+
+let results = SAMPLES.map(function(SIZE) {
+  let stats = {}
+
+  repo.addDomain('pixels', {
+    getInitialState() {
+      return {}
+    },
+    rotateHue(n) {
+      return isNaN(n) ? 0 : n + 5
+    },
+    advance(state, { x, y }) {
+      return update(state, [x, y], this.rotateHue)
```

Comment:
We should improve the docs on this. `get`, `set`, and `update` accept a key path. This could be:

1. A string: `a.b.c`
2. An array of string keys `['a', 'b', 'c']`